packages/@react-aria/utils/src/useId.ts

```diff
@@ -87,18 +88,20 @@ export function mergeIds(idA: string, idB: string): string {
 /**
  * Used to generate an id, and after render, check if that id is rendered so we know
  * if we can use it in places such as labelledby.
+ * @param depArray - When to recalculate if the id is in the DOM.
  */
-export function useSlotId(): string {
+export function useSlotId(depArray: any[] = []): string {
   let id = useId();
-  let [resolvedId, setResolvedId] = useState(id);
-  useLayoutEffect(() => {
-    let setCurr = idsUpdaterMap.get(id);
```

Comment:
unsure if i need this setCurr, i think it was left over from an earlier change and doesn't contribute anything